prog/types.go

```diff
@@ -81,6 +82,62 @@ func (f *Field) Dir(def Dir) Dir {
 	return def
 }
 
+type ValueExpression interface {
```

Comment:
I would call it just Expression.
We don't have other expressions and it's assumed expressions evaluate to values.